scripts/generate_source.py

```diff
@@ -33,22 +34,39 @@
 def main(argv):
     parser = argparse.ArgumentParser(description='Generate source code for this repository')
     parser.add_argument('registry', metavar='REGISTRY_PATH', help='path to the Vulkan-Headers registry directory')
+    parser.add_argument('--api', help='generate extensions for the api provided vulkan or vulkansc', default='vulkan')
     group = parser.add_mutually_exclusive_group()
     group.add_argument('-i', '--incremental', action='store_true', help='only update repo files that change')
     group.add_argument('-v', '--verify', action='store_true', help='verify repo files match generator output')
     args = parser.parse_args(argv)
 
+    vksc_filenames =['vk_loader_extensions.h',
+                     'vk_loader_extensions.c',
+                     'vk_dispatch_table_helper.h',
+                     'vk_layer_dispatch_table.h',
+                     'loader_generated_header_version.cmake']
+
+    vk_filenames   =['vk_dispatch_table_helper.h',
+                     'vk_layer_dispatch_table.h',
+                     'vk_loader_extensions.h',
+                     'vk_loader_extensions.c',
+                     'vk_object_types.h',
+                     'loader_generated_header_version.cmake']
+
```

Comment:
WHy have all this duplication?  Why not a core filenames list and then a desktop list that is added to the first list?  You can even add it in the 'if' below if you use != 'vulkansc'...